src/controllers/urls.go

```diff
@@ -53,6 +54,10 @@ var (
 		status:  fiber.ErrForbidden.Code,
 		message: "this shortURL is not allowed to be created",
 	}
+	MaxRetriesReachedError = &UrlError{
+		status:  fiber.ErrInternalServerError.Code,
+		message: "max number of attempts to generate a URL reached. Please try after some time",
```

Comment:
so this is not a good error to give the client, because the client is technically not "retrying" these many times. It is our code's internal recursion causing it. 

A 508 `StatusLoopDetected`  error with message "server exhausted attempts to generate a unique shortUrl, please try again" would be better

Reply:
Thanks for the suggestion. Have updated the message.